libraries/ESP8266MQTTClient/src/URIParser.h

```diff
@@ -0,0 +1,28 @@
+#ifndef URIParser_h_
+#define URIParser_h_
+typedef struct{
+	char *scheme;               /* mandatory */
+	char *host;                 /* mandatory */
+	char *port;                 /* optional */
+	char *path;                 /* optional */
+	char *query;                /* optional */
+	char *fragment;             /* optional */
+	char *username;             /* optional */
+	char *password;             /* optional */
+} parsed_url;
```

Comment:
this should end on _t and maybe there could be a better name? what is the result? URI or URL actually (i kinda get confused with both).